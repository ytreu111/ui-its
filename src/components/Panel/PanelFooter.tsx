import React, { FC } from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "styles";
import { Styled } from "./style/PanelFooter.styled";
import { IPanelFooterProps } from "./model/PanelFooter.model";

const {
  PanelFooterWrap,
} = Styled;

const PanelFooter: FC<IPanelFooterProps> = (
  {
    style,
    className,
    children,
  }) => {

  return (
    <ThemeProvider theme={theme}>
      <PanelFooterWrap style={style} className={className}>
        {children}
      </PanelFooterWrap>
    </ThemeProvider>
  );
}

export default PanelFooter;