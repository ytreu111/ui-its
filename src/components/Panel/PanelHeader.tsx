import React, { FC } from "react";
import theme from "styles/theme/theme";
import { ThemeProvider } from "styled-components";
import { Styled } from "./style/PanelHeader.styled";
import { IPanelHeaderProps } from "./model/PanelHeader.model";

const {
  PanelHeaderWrap,
} = Styled;

const PanelHeader: FC<IPanelHeaderProps> = (
  {
    style,
    className,
    children,
  }) => {

  return (
    <ThemeProvider theme={theme}>
      <PanelHeaderWrap className={className} style={style}>
        {children}
      </PanelHeaderWrap>
    </ThemeProvider>
  );
}

export default PanelHeader;