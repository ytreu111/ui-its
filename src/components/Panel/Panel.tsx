import React, { FC, useMemo } from "react";
import { ThemeProvider } from "styled-components";

import { IPanelProps } from "./model/Panel.model";
import { Styled } from "./style/Panel.styled";
import theme from "styles/theme/theme";
import PanelHeader from "./PanelHeader";
import PanelFooter from "./PanelFooter";

const {
  PanelWrap,
  PanelContentWrap,
} = Styled;

const Panel: FC<IPanelProps> = (
  {
    children,
    headerContent,
    headerStyle,
    headerClass,
    headerComponent,
    footerContent,
    footerStyle,
    footerClass,
    footerComponent,
    contentClass,
    contentStyle,
    ...props
  },
) => {

  const header = useMemo(() => {
    if (headerComponent) {
      return headerComponent;
    } else if (headerContent) {
      return <PanelHeader className={headerClass} style={headerStyle}>{headerContent}</PanelHeader>
    }

    return null;
  }, [headerClass, headerComponent, headerContent, headerStyle]);

  const footer = useMemo(() => {
    if (footerComponent) {
      return footerComponent;
    } else if (footerContent) {
      return <PanelFooter className={footerClass} style={footerStyle}>{footerContent}</PanelFooter>
    }

    return null;
  }, [footerClass, footerComponent, footerContent, footerStyle]);

  return (
    <ThemeProvider theme={theme}>
      <PanelWrap {...props}>
        {header}
        <PanelContentWrap className={contentClass} style={contentStyle}>
          {children}
        </PanelContentWrap>
        {footer}
      </PanelWrap>
    </ThemeProvider>
  );
}

export default Panel;