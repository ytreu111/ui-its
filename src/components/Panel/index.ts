import { FC } from "react";
import InternalPanel from "./Panel";
import { IPanelProps } from "./model/Panel.model";
import PanelHeader from "./PanelHeader";
import PanelFooter from "./PanelFooter";

export interface CompoundedComponent extends FC<IPanelProps> {
  Header: typeof PanelHeader,
  Footer: typeof PanelFooter,
}

const Panel = InternalPanel as CompoundedComponent;
Panel.Header = PanelHeader;
Panel.Footer = PanelFooter;

export type { IPanelProps } from './model/Panel.model'
export type { IPanelHeaderProps } from './model/PanelHeader.model'
export type { IPanelFooterProps } from './model/PanelFooter.model'
export default Panel