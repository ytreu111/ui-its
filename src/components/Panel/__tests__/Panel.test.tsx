import { render, prettyDOM, screen } from '@testing-library/react';
import Panel from '../Panel';

const customComponent = <div>panel</div>;

describe('Компонент Panel', () => {
  test('Шапка', () => {
    const { rerender, baseElement } = render(<Panel headerComponent={customComponent} />);
    expect(screen.getByText('panel')).toBeInTheDocument();

    rerender(
      <Panel
        headerContent={'header-1'}
        headerClass={'header'}
        headerStyle={{ color: 'black' }}
      />,
    );
    expect(screen.getByText('header-1')).toBeInTheDocument();
    expect(screen.getByText('header-1')).toHaveClass('header');
    expect(screen.getByText('header-1')).toHaveStyle('color: black');

    rerender(<Panel />);
    expect(prettyDOM(baseElement)).toMatchSnapshot();
  });
  test('Футер', () => {
    const { rerender, baseElement } = render(<Panel footerComponent={customComponent} />);
    expect(screen.getByText('panel')).toBeInTheDocument();

    rerender(
      <Panel
        footerContent={'footer-1'}
        footerClass={'footer'}
        footerStyle={{ color: 'black' }}
      />,
    );
    expect(screen.getByText('footer-1')).toBeInTheDocument();
    expect(screen.getByText('footer-1')).toHaveClass('footer');
    expect(screen.getByText('footer-1')).toHaveStyle('color: black');

    rerender(<Panel />);
    expect(prettyDOM(baseElement)).toMatchSnapshot();
  });
  test('Контент', () => {
    const { baseElement } = render(
      <Panel contentClass={'content'} contentStyle={{ color: 'black' }}>
        {customComponent}
      </Panel>,
    );
    expect(prettyDOM(baseElement)).toMatchSnapshot();
  });
});