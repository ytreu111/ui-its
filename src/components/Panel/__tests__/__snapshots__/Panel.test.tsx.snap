// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`–ö–æ–º–ø–æ–Ω–µ–Ω—Ç Panel –ö–æ–Ω—Ç–µ–Ω—Ç 1`] = `
"[36m<body>[39m
  [36m<div>[39m
    [36m<div[39m
      [33mclass[39m=[32m\\"sc-bdvvtL bxZgRB\\"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m\\"sc-gsDKAQ eQfDAi content\\"[39m
        [33mstyle[39m=[32m\\"color: black;\\"[39m
      [36m>[39m
        [36m<div>[39m
          [0mpanel[0m
        [36m</div>[39m
      [36m</div>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</body>[39m"
`;

exports[`–ö–æ–º–ø–æ–Ω–µ–Ω—Ç Panel –§—É—Ç–µ—Ä 1`] = `
"[36m<body>[39m
  [36m<div>[39m
    [36m<div[39m
      [33mclass[39m=[32m\\"sc-bdvvtL bxZgRB\\"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m\\"sc-gsDKAQ eQfDAi\\"[39m
      [36m/>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</body>[39m"
`;

exports[`–ö–æ–º–ø–æ–Ω–µ–Ω—Ç Panel –®–∞–ø–∫–∞ 1`] = `
"[36m<body>[39m
  [36m<div>[39m
    [36m<div[39m
      [33mclass[39m=[32m\\"sc-bdvvtL bxZgRB\\"[39m
    [36m>[39m
      [36m<div[39m
        [33mclass[39m=[32m\\"sc-gsDKAQ eQfDAi\\"[39m
      [36m/>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</body>[39m"
`;
